### PR TITLE
Refactors sound groups to use decls instead of a massive switch with dozens of global lists.

### PR DESCRIPTION
**Pull request by NonQueueingMatt**
As per title. This also fixes some sounds that didn't play previously, since they had wrong names.

### CODE REVIEW

**Comment by NonQueueingMatt** [comment]
!review

File: code/game/sound.dm [hunk 1]
```diff
@@ -469,58 +219,335 @@ var/list/gauss_fire_sound = list(
...
+	sounds = list(
+		'sound/effects/bodyfall1.ogg',
+		'sound/effects/bodyfall2.ogg',
+		'sound/effects/bodyfall3.ogg',
+		'sound/effects/bodyfall4.ogg'
+	)
+
+/decl/sound_category/bodyfall_machine_sound
+	sounds = list(
+	'sound/effects/bodyfall_machine1.ogg',
```
Comment (mikomyazaki) [inline code/game/sound.dm:494]: Indentation

File: code/game/sound.dm [hunk 2]
```diff
@@ -469,58 +219,335 @@ var/list/gauss_fire_sound = list(
...
+	'sound/effects/bodyfall_machine2.ogg'
+	)
+/decl/sound_category/bulletflyby_sound
+	sounds = list(
+	'sound/effects/bulletflyby1.ogg',
```
Comment (mikomyazaki) [inline code/game/sound.dm:499]: and here

File: code/game/sound.dm [hunk 3]
```diff
@@ -469,58 +219,335 @@ var/list/gauss_fire_sound = list(
...
-			if ("drillhit") soundin = pick(drillhit_sound)
-			if ("wield_generic") soundin = pick(wield_generic_sound)
-			if ("equip_sword") soundin = pick(sword_equip_sound)
-			if ("pickup_sword") soundin = pick(sword_pickup_sound)
-			if ("gauss_fire") soundin = pick(gauss_fire_sound)
-	return soundin
+/proc/get_sfx(var/sound_category)
+	var/decl/sound_category/SC = decls_repository.get_decl(sound_category)
+	if(!istype(SC))
+		crash_with("Non-decl path in get_sfx: [sound_category]")
```
Comment (Geevies) [inline code/game/sound.dm:233]:
```suggestion
		CRASH("Non-decl path in get_sfx: [sound_category]")
```

File: code/game/sound.dm [hunk 4]
```diff
@@ -469,58 +219,335 @@ var/list/gauss_fire_sound = list(
...
+		return
```
Comment (Geevies) [inline code/game/sound.dm:234]:
```suggestion
```

File: code/game/sound.dm (hunk 3 above)
Comment (Geevies) [reply code/game/sound.dm:233]: image

**Pull request merged**